### PR TITLE
feat(feature-section): update to v2

### LINKED ISSUE #10852: [Component] Feature section

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

Closes #10852 

### Description

Update feature section to v2
### Changelog

**Changed**

- deprecated feature-section-card-link in favor for `card` with link variation
- readme and examples updated

### CODE REVIEW

File: packages/styles/scss/components/feature-section/_feature-section.scss [hunk 1]
```diff
@@ -44,6 +60,10 @@
     background-color: $layer-01;
     display: block;
 
+    div[class^='#{$prefix}--col'] {
+      box-sizing: border-box;
+    }
+
```
Comment (IgnacioBecerra) [inline packages/styles/scss/components/feature-section/_feature-section.scss:66]: Maybe it's about style specificity, but I'm curious as to why it was done like this over `.#{$prefix}--col`?

**Review by IgnacioBecerra — APPROVED** [review]
This looks good to me! Just a small question below:

File: packages/styles/scss/components/feature-section/_feature-section.scss (hunk 1 above)
Comment (ariellalgilmore) [reply packages/styles/scss/components/feature-section/_feature-section.scss:66]: i am targetting this line: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/b5ecd5f42dd85c241909d80a361cd238c58cdb94/packages/web-components/src/components/feature-section/feature-section.ts#L70 , so just thought I would target any classes starting with `.#{$prefix}--col`

**Review by oliviaflory — APPROVED** [review]
😭 sobbing it's so perfect!

**Pull request merged**